fvm/evm/precompiles/abi.go

```diff
@@ -0,0 +1,223 @@
+package precompiles
+
+import (
+	"encoding/binary"
+	"errors"
+	"math/big"
+
+	gethCommon "github.com/ethereum/go-ethereum/common"
+)
+
+// This package provides fast and effient
+// utilities needed for abi encoding and decoding
+// encodings are mostly used for testing purpose
+// if more complex encoding and decoding is needed please
+// use the abi package and pass teh ABIs, though
+// that has a performance overhead.
+const (
+	FixedSizeUnitDataReadSize = 32
+	Bytes4DataReadSize        = 4
+	Bytes8DataReadSize        = 8
+	Bytes32DataReadSize       = 32
+	Uint64ByteSize            = 8
+
+	EncodedBoolSize    = FixedSizeUnitDataReadSize
+	EncodedAddressSize = FixedSizeUnitDataReadSize
+	EncodedBytes32Size = FixedSizeUnitDataReadSize
+	EncodedBytes4Size  = FixedSizeUnitDataReadSize
+	EncodedBytes8Size  = FixedSizeUnitDataReadSize
+	EncodedUint64Size  = FixedSizeUnitDataReadSize
+	EncodedUint256Size = FixedSizeUnitDataReadSize
+)
+
+var ErrInputDataTooSmall = errors.New("input data is too small for decoding")
+var ErrBufferTooSmall = errors.New("buffer too small for encoding")
+var ErrDataTooLarge = errors.New("input data is too large for encoding")
+
+// ReadAddress reads an address from the buffer at index
+func ReadAddress(buffer []byte, index int) (gethCommon.Address, error) {
```

Comment:
Should the index be uint? the negative value provided for the index would panic

Reply:
I'm not really worried about the uint aspect its and internal method, using uint then requires continuous conversion to int for slicing.